spec/controllers/blackouts_controller_spec.rb

```diff
@@ -116,53 +116,112 @@
         end
       end
     end
+
     context 'POST create' do
-      context 'with correct params' do
-        before do
-          @attributes = FactoryGirl.attributes_for(:blackout)
-          post :create, blackout: @attributes
-        end
+      shared_examples_for 'creates blackout' do |attributes|
+        before { post :create, blackout: attributes }
+
         it 'should create the new blackout' do
           expect(Blackout.find(assigns(:blackout).id)).not_to be_nil
         end
         it 'should pass the correct params' do
-          expect(assigns(:blackout)[:notice]).to eq(@attributes[:notice])
+          expect(assigns(:blackout)[:notice]).to eq(attributes[:notice])
           expect(assigns(:blackout)[:start_date]).to\
-            eq(@attributes[:start_date])
-          expect(assigns(:blackout)[:end_date]).to eq(@attributes[:end_date])
+            eq(attributes[:start_date])
+          expect(assigns(:blackout)[:end_date]).to eq(attributes[:end_date])
           expect(assigns(:blackout)[:blackout_type]).to\
-            eq(@attributes[:blackout_type])
+            eq(attributes[:blackout_type])
         end
         it { is_expected.to redirect_to(blackout_path(assigns(:blackout))) }
         it { is_expected.to set_flash }
       end
-      context 'with incorrect params' do
+
+      shared_examples_for 'does not create blackout' do |attributes|
```

Comment:
Sure. On second thought, there's no reason why these all can't just use the same start and end dates (and we'll just modify the reservation dates to make it work), so we don't need to pass anything. I'll update and re-push.